src/main/java/com/eerussianguy/blazemap/feature/maps/Coordination.java

```diff
@@ -14,8 +14,13 @@ public void calculate(int mouseX, int mouseY, int beginX, int beginZ, double zoo
         mousePixelX = mouseX;
         mousePixelY = mouseY;
 
-        blockX = (int) (beginX + (mouseX / zoom));
-        blockZ = (int) (beginZ + (mouseY / zoom));
+        if (zoom >= 1) {
+            blockX = (beginX + (mouseX / (int)zoom));
+            blockZ = (beginZ + (mouseY / (int)zoom));
```

Comment:
Making this an int division seems to stop the block/chunk/region boxes from jumping around. And once you're far enough out to reach the other condition the jumping wasn't even noticeable (at least to my eyes).